mlrun/model_monitoring/db/_schedules.py

```diff
@@ -149,6 +162,80 @@ def get_min_timestamp(self) -> Optional[int]:
         return min(self._schedules.values(), default=None)
 
 
+class ModelMonitoringSchedulesFileChief(ModelMonitoringSchedulesFileBase):
+    def __init__(self, project: str) -> None:
+        """
+        Initialize applications monitoring schedules file object.
+        The JSON file stores a dictionary of registered application name as key and Unix timestamp as value.
+        When working with the schedules data, use this class as a context manager to read and write the data.
+
+        :param project:     The project name.
+        """
+        # `self._item` is the persistent version of the monitoring schedules.
+        super().__init__()
+        self._item = mlrun.model_monitoring.helpers.get_monitoring_schedules_chief_data(
+            project=project
+        )
+        self._post_init()
+
+    def _post_init(self):
+        self._path = self._item.url
+        self._fs = self._item.store.filesystem
+        # `self._schedules` is an in-memory copy of the DB for all the applications for
+        # the same model endpoint.
+        self._schedules: dict[str, dict[str, int]] = {}
+        # Does `self._schedules` hold the content of `self._item`?
+        self._open_schedules = False
+
+    def get_endpoint_last_request(self, endpoint_uid: str) -> Optional[int]:
+        self._check_open_schedules()
+        if endpoint_uid in self._schedules:
+            return self._schedules[endpoint_uid].get(FileTargetKind.LAST_REQUEST)
+        else:
+            return None
+
+    def update_endpoint_last_request(self, endpoint_uid: str, timestamp: int) -> None:
+        self._check_open_schedules()
+        if endpoint_uid in self._schedules:
+            self._schedules[endpoint_uid][FileTargetKind.LAST_REQUEST] = timestamp
+        else:
+            self._schedules[endpoint_uid] = {
+                FileTargetKind.LAST_ANALYZED: None,
+                FileTargetKind.LAST_REQUEST: timestamp,
+            }
+
+    def update_endpoint_last_analyzed(self, endpoint_uid: str, timestamp: int) -> None:
+        self._check_open_schedules()
+        if endpoint_uid in self._schedules:
+            self._schedules[endpoint_uid][FileTargetKind.LAST_ANALYZED] = timestamp
+        else:
+            self._schedules[endpoint_uid] = {
+                FileTargetKind.LAST_ANALYZED: timestamp,
+                FileTargetKind.LAST_REQUEST: None,
+            }
```

Comment:
Why not to create one method to update both? 
Do we have a scenario which in we update only one of them?